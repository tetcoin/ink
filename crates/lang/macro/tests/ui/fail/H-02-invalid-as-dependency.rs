use pro_lang as pro;

#[pro::contract(compile_as_dependency = "yes")]
mod invalid_as_dependency {
    #[pro(storage)]
    pub struct InvalidAsDependency {}

    impl InvalidAsDependency {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message)]
        pub fn message(&self) {}
    }
}

fn main() {}
