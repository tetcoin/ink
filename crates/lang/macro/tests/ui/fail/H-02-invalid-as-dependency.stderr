error: expected a bool literal for `compile_as_dependency` pro! config argument
 --> $DIR/H-02-invalid-as-dependency.rs:3:17
  |
3 | #[pro::contract(compile_as_dependency = "yes")]
  |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
